### PR TITLE
feat: LOPS-334: Support AWS/Azure Dataset Storage

### PR DESCRIPTION
**Pull request by KonstantinKorotaev**
### PR fulfills these requirements
- [X] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [X] Tests for the changes have been added/updated (for bug fixes/features)
- [X] Docs have been added/updated (for bug fixes/features)
- [X] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [X] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [X] Backend (API)
- [ ] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [X] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [X] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_
Storages

### CODE REVIEW

File: label_studio/tests/utils.py [hunk 1]
```diff
@@ -146,6 +146,8 @@ def upload_blob(self, string, overwrite):
             print(f'String {string} uploaded to bucket {self.container_name}')
         def generate_signed_url(self, **kwargs):
             return f'https://storage.googleapis.com/{self.container_name}/{self.key}'
+        def content_as_text(self):
```
Comment (hakan458) [inline label_studio/tests/utils.py:149]: What is this? Just used for testing?

File: label_studio/tests/utils.py [hunk 2]
```diff
@@ -170,6 +172,9 @@ def get_container_properties(self, **kwargs):
                 version='1.0.0'
             )
 
+        def download_blob(self, key):
```
Comment (hakan458) [inline label_studio/tests/utils.py:175]: Just for tests?

File: label_studio/tests/utils.py (hunk 1 above)
Comment (KonstantinKorotaev) [reply label_studio/tests/utils.py:149]: Yes, it's a mock

File: label_studio/tests/utils.py (hunk 2 above)
Comment (KonstantinKorotaev) [reply label_studio/tests/utils.py:175]: Yes, it's a mock

**Review by hakan458 — APPROVED** [review]

**Review by MihajloHoma — APPROVED** [review]

**Pull request merged**